src/modules/private-share-folder/private-sharing.usecase.ts

```diff
@@ -2,12 +2,44 @@ import { Injectable } from '@nestjs/common';
 import { Folder } from '../folder/folder.domain';
 import { User } from '../user/user.domain';
 import { SequelizePrivateSharingRepository } from './private-sharing.repository';
+import { SequelizeUserRepository } from '../user/user.repository';
+import { SequelizeFolderRepository } from '../folder/folder.repository';
 
+export class InvalidOwnerError extends Error {
+  constructor() {
+    super('You are not the owner of this folder');
+  }
+}
 @Injectable()
 export class PrivateSharingUseCase {
   constructor(
     private privateSharingRespository: SequelizePrivateSharingRepository,
+    private userRespository: SequelizeUserRepository,
+    private folderRespository: SequelizeFolderRepository,
   ) {}
+  async grantPrivileges(
+    owner: User,
+    userUuid: string,
+    privateFolderId: string,
```

Comment:
Could be a good idea to reference this type by the entity directly so in case the type of the entity field changes, you do not have ti change it on the entire codebase one by one. So instead of:

`privateFolderId: string`

you have:

`privateFolderId: Folder['uuid'] `

or even better:

`PrivateSharingFolder['folderId']`